formula/src/main/java/com/instacart/formula/Update.kt

```diff
@@ -1,41 +1,13 @@
 package com.instacart.formula
 
-class Update<Message>(
-    val key: Any,
-    val stream: Stream<Message>,
-    initial: (Message) -> Unit
-) {
-
-    internal var handler: (Message) -> Unit = initial
-    internal var cancelable: Cancelable? = null
-
-    internal fun start() {
-        cancelable = stream.start() { message ->
-            handler.invoke(message)
-        }
-    }
-
-    internal fun tearDown() {
-        cancelable?.cancel()
-        cancelable = null
-    }
-
-    override fun equals(other: Any?): Boolean {
-        if (this === other) return true
-        if (javaClass != other?.javaClass) return false
-
-        other as Update<*>
-
-        if (key != other.key) return false
-
-        return true
-    }
+/**
+ * An update function takes an [Event] and calculates the [Transition] that should happen. A
+ * transition can contain a new [state][State] object which would trigger a [Formula.evaluate]
+ * and/or [Effects] that should be performed. You can return [Transition.None] if nothing
+ * should happen.
+ */
+fun interface Update<State, Event> {
```

Comment:
I'm wondering how to better unify this with the `Transition` type. Maybe this should be changed to 
```kotlin
fun interface Transition<State, Event> {
    sealed class Result {
       /**
         * Stateful transition.
         *
         * @param state New state
         * @param effects Optional effects such as parent callbacks, logging, db writes,
         * network requests, etc.
         */
        data class Stateful<State>(val state: State, override val effects: Effects? = null) : Result<State>()

        /**
         * Only effects are emitted as part of this transition
         *
         * @param effects Effects such as parent callbacks, logging, db writes, network requests, etc.
         */
        data class OnlyEffects(override val effects: Effects) : Result<Nothing>()

        /**
         * Nothing happens in this transition.
         */
        object None : Result<Nothing>() {
            override val effects: Effects? = null
        }
   }

   fun TransitionContext.create(event: Event): Result<State>
}
```

Reply:
Yeah, having an `Update` that produces a `Transition` is a bit confusing, would be better to not have that extra name I think.

Reply:
Does `Update` and `Update.Result` or `Transition` and `Transition.Result` seem better? On one hand, `Update` corresponds to `MVU` naming, however many folks like the name `Transition` (including me).

Reply:
I'm moving this change out of this PR.